### DISCUSSION TITLE
Auto generation of c# class model from Ontology

### DISCUSSION THREAD

**Discussion by jiatao99** [dotnetrdf/dotnetrdf#594] [General]
Is it possible to auto generate c# model class from ontology schma ttl file?

**Comment by giacomociti** [comment]
You may have a look at my little [project](https://github.com/giacomociti/iride). It is fairly unconventional but I’m successfully using it in a production C# codebase

**Comment by joephayes** [comment]
You can also take a look at [OLGA](https://github.com/EcoStruxure/OLGA)

**Comment by langsamu** [comment]
[This library](https://github.com/ukparliament/Mapping) by @hellikopter does just that.
Not documented but might help you.

**Comment by jiatao99** [comment]
> [This library](https://github.com/ukparliament/Mapping) by @hellikopter does just that.
> Not documented but might help you.

Thanks. Will check it

**Comment by langsamu** [comment]
> > [This library](https://github.com/ukparliament/Mapping) by @hellikopter does just that.
> > Not documented but might help you.
> 
> Thanks. Will check it

That library uses CodeDom to translate an OWL ontology to C#/DLL.
Take a look at the [runner](https://github.com/ukparliament/Mapping/blob/master/Parliament.Ontology.Mapper.Compiler/Program.cs), [CodeDom specialisations](https://github.com/ukparliament/Mapping/tree/master/Parliament.Ontology/ModelCodeDom) and [MSBuild generator task](https://github.com/ukparliament/Mapping/blob/master/Parliament.Model/Parliament.Model.csproj#L54).

You might also benefit from dotNetRDF's [ontology abstractions](https://github.com/dotnetrdf/dotnetrdf/wiki/UserGuide-Ontology-API) if you havent found them yet, as well as its [dynamic RDF mapping capability](https://github.com/dotnetrdf/dotnetrdf/wiki/UserGuide-Dynamic-API).

**Comment by kal** [comment]
I'm going to convert this question to a dicussion as I think it might be helpful for others to find later.
